service/src/main/kotlin/fi/espoo/evaka/daycare/DaycareQueries.kt

```diff
@@ -86,18 +86,25 @@ data class DaycareFields(
                 }
             }
         }
-    }
-
-    fun validateClosingDate(tx: Database.Read, daycareId: DaycareId) {
-        val lastPlacementDate = tx.getLastPlacementDate(daycareId)
-        if (closingDate != null && lastPlacementDate != null && lastPlacementDate > closingDate) {
-            throw BadRequest("Closing date cannot be before the last placement date")
+        if (openingDate != null && closingDate != null && closingDate.isBefore(openingDate)) {
+            throw BadRequest("Closing date cannot be before opening date")
         }
     }
 
     companion object {}
 }
 
+fun validateUnitClosingDate(tx: Database.Read, unitId: DaycareId, closingDate: LocalDate) {
+    val openingDate = tx.getDaycare(unitId)?.openingDate
+    if (openingDate != null && closingDate.isBefore(openingDate)) {
+        throw BadRequest("Closing date cannot be before opening date")
+    }
```

Comment:
this could work incorrectly when doing full update in case the openingDate is also updated